nemo_rl/models/generation/vllm/routing_policy.py

```
@@ -0,0 +1,246 @@
+    """
+    if config is None:
+        return RoundRobinRoutingPolicy(dp_leader_worker_indices)
+
+    policy_type = config.get("type", "round_robin")
+
+    if policy_type == "round_robin":
+        return RoundRobinRoutingPolicy(dp_leader_worker_indices)
+    elif policy_type == "cache_aware":
+        router_class_path = config.get("router_class")
+        if not router_class_path:
+            raise ValueError("CacheAwareRoutingPolicy requires 'router_class' in config")
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)